### PR TITLE
Redeploy when a secret/configmap from top-level config changes

### PR DESCRIPTION
**Pull request by stlaz**
Redeploys the osin server if any of the configMaps and secrets in the top-level config changed.

Manually tested.

### CODE REVIEW

File: pkg/operator2/operator.go [hunk 1]
```diff
@@ -200,35 +208,36 @@ func (c *authOperator) handleSync(operatorConfig *operatorv1.Authentication) err
...
 	cliConfig, _, err := resourceapply.ApplyConfigMap(c.configMaps, c.recorder, expectedCLIconfig)
 	if err != nil {
 		return err
 	}
+	resourceVersions = append(resourceVersions, cliConfig.GetResourceVersion())
 
 	// deployment, have RV of all resources
 	// TODO use ExpectedDeploymentGeneration func
-	// TODO manually get RV of all the config maps and secrets in syncData
 	// TODO we also need the RV for the serving-cert secret (servingCertName)
```
Comment (stlaz) [inline pkg/operator2/operator.go]: Isn't the CA bundle enough?

Comment (ericavonb) [reply pkg/operator2/operator.go]: The serving-cert could be rotated independently of the CA.

**Pull request merged**